packages/twenty-server/src/metadata/field-metadata/composite-types/currency.composite-type.ts

```diff
@@ -2,10 +2,22 @@ import { ObjectMetadataInterface } from 'src/metadata/field-metadata/interfaces/
 import { FieldMetadataInterface } from 'src/metadata/field-metadata/interfaces/field-metadata.interface';
 
 import { FieldMetadataType } from 'src/metadata/field-metadata/field-metadata.entity';
+import { generateTargetColumnMap } from 'src/metadata/field-metadata/utils/generate-target-column-map.util';
 
 export const currencyFields = (
   fieldMetadata?: FieldMetadataInterface,
 ): FieldMetadataInterface[] => {
+  const targetColumnMap = fieldMetadata
+    ? generateTargetColumnMap(
+        fieldMetadata.type,
+        fieldMetadata.isCustom ?? false,
+        fieldMetadata.name,
+      )
+    : {
+        amountMicros: 'amountMicros',
```

Comment:
So basically if fieldMetadata is here, this is used to map for pg_graphql. 

If fieldMetadata is not here, this is for the return of the schema builder so we don't want the name of the field and have a clean 
budget {
  amountMicros
  currencyCode
}

I'm wondering if it was a good idea to use the same code for 2 things that seem quite different?
Wdyt?